explicit_acl should not be triggered for extensions adding protocol conformance

### New Issue Checklist

- [X] Updated SwiftLint to the latest version
- [X] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

### Describe the bug
Swiftlint tells me that I need to add an access control level for an extension that adds protocol conformance. However, the Swift compiler won't allow me to do that -- when I try to build I get
`error: 'public' modifier cannot be used with extensions that declare protocol conformances`. (Note I get an equivalent error if I use any other ACL, `private`, `internal`, etc.) 

This can be reproduced in a directory containing just the following two files:

**lint.swift**
```swift
public struct MyStruct {}
extension MyStruct: Equatable {}
```

**.swiftlint.yml**
```yml
opt_in_rules:
  - explicit_acl
```

##### Complete output when running SwiftLint, including the stack trace and command used

```bash
kaitlinmahar@kaitlin:~/lint-test$ swiftlint lint
Loading configuration from '.swiftlint.yml'
Linting Swift files at paths
Linting 'lint.swift' (1/1)
/Users/kaitlinmahar/lint-test/lint.swift:2:1: warning: Explicit ACL Violation: All declarations should specify Access Control Level keywords explicitly. (explicit_acl)
Done linting! Found 1 violation, 0 serious in 1 file.
```

If I change the file to:
```swift
public struct MyStruct {}
public extension MyStruct: Equatable {}
```

Swiftlint is satisfied -- 
```bash
kaitlinmahar@kaitlin:~/lint-test$ swiftlint lint
Loading configuration from '.swiftlint.yml'
Linting Swift files at paths
Linting 'lint.swift' (1/1)
Done linting! Found 0 violations, 0 serious in 1 file.
```

But the compiler is not:
```bash
kaitlinmahar@kaitlin:~/lint-test$ swift lint.swift
lint.swift:2:1: error: 'public' modifier cannot be used with extensions that declare protocol conformances
public extension MyStruct: Equatable {}
^~~~~~~
```

### Environment

* SwiftLint version (run `swiftlint version` to be sure)? `0.31.0`
* Installation method used (Homebrew, CocoaPods, building from source, etc)? Homebrew
* Which Xcode version are you using (check `xcode-select -p`)? xcode-select version 2354. However, I'm using Swift 4.2 installed via [swiftenv](http://swiftenv.fuller.li/en/latest/).
```bash
kaitlinmahar@kaitlin:~/lint-test$ swift --version
Apple Swift version 4.2 (swift-4.2-RELEASE)
Target: x86_64-apple-darwin18.2.0
```
I also switched to Swift 5.0 via swiftenv and tested, same results.

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -72,6 +72,11 @@
   [Keith Smiley](https://github.com/keith)
   [#2703](https://github.com/realm/SwiftLint/issues/2703)
 
+* Fix false positives on `explicit_acl` and `explicit_top_level_acl` rules when
+  declaring extensions that add protocol conformances with Swift 5.  
+  [Marcelo Fabri](https://github.com/marcelofabri)
+  [#2705](https://github.com/realm/SwiftLint/issues/2705)
+
 ## 0.31.0: Busy Laundromat
 
 #### Breaking

--- a/Rules.md
+++ b/Rules.md
@@ -5728,6 +5728,10 @@ internal protocol A {
 internal class A { deinit {} }
 ```
 
+```swift
+extension A: Equatable {}
+```
+
 </details>
 <details>
 <summary>Triggering Examples</summary>
@@ -6032,6 +6036,10 @@ class Foo {}
 ```swift
 internal func a() {}
 
+```
+
+```swift
+extension A: Equatable {}
 ```
 
 </details>

--- a/Source/SwiftLintFramework/Rules/Idiomatic/ExplicitACLRule.swift
+++ b/Source/SwiftLintFramework/Rules/Idiomatic/ExplicitACLRule.swift
@@ -39,7 +39,8 @@ public struct ExplicitACLRule: OptInRule, ConfigurationProviderRule, AutomaticTe
               var b: Int
             }
             """,
-            "internal class A { deinit {} }"
+            "internal class A { deinit {} }",
+            "extension A: Equatable {}"
         ],
         triggeringExamples: [
             "enum A {}\n",
@@ -60,9 +61,18 @@ public struct ExplicitACLRule: OptInRule, ConfigurationProviderRule, AutomaticTe
 
     private func offsetOfElements(from elements: [SourceKittenElement], in file: File,
                                   thatAreNotInRanges ranges: [NSRange]) -> [Int] {
+        let extensionKinds: Set<SwiftDeclarationKind> = [.extension, .extensionClass, .extensionEnum,
+                                                         .extensionProtocol, .extensionStruct]
+
         return elements.compactMap { element in
             guard let typeOffset = element.offset else {
                 return nil
+            }
+
+            guard let kind = element.kind.flatMap(SwiftDeclarationKind.init(rawValue:)),
+                case let isConformanceExtension = extensionKinds.contains(kind) && !element.inheritedTypes.isEmpty,
+                !isConformanceExtension else {
+                    return nil
             }
 
             // find the last "internal" token before the type

--- a/Source/SwiftLintFramework/Rules/Idiomatic/ExplicitTopLevelACLRule.swift
+++ b/Source/SwiftLintFramework/Rules/Idiomatic/ExplicitTopLevelACLRule.swift
@@ -18,7 +18,8 @@ public struct ExplicitTopLevelACLRule: OptInRule, ConfigurationProviderRule, Aut
             "internal enum A {\n enum B {}\n}",
             "internal final class Foo {}",
             "internal\nclass Foo {}",
-            "internal func a() {}\n"
+            "internal func a() {}\n",
+            "extension A: Equatable {}"
         ],
         triggeringExamples: [
             "enum A {}\n",
@@ -30,8 +31,18 @@ public struct ExplicitTopLevelACLRule: OptInRule, ConfigurationProviderRule, Aut
     )
 
     public func validate(file: File) -> [StyleViolation] {
+        let extensionKinds: Set<SwiftDeclarationKind> = [.extension, .extensionClass, .extensionEnum,
+                                                         .extensionProtocol, .extensionStruct]
+
         // find all top-level types marked as internal (either explictly or implictly)
         let internalTypesOffsets = file.structure.dictionary.substructure.compactMap { element -> Int? in
+            // ignore extensions that declare protocol conformance
+            guard let kind = element.kind.flatMap(SwiftDeclarationKind.init(rawValue:)),
+                case let isConformanceExtension = extensionKinds.contains(kind) && !element.inheritedTypes.isEmpty,
+                !isConformanceExtension else {
+                    return nil
+            }
+
             if element.accessibility.flatMap(AccessControlLevel.init(identifier:)) == .internal {
                 return element.offset
             }

